Fixes #1104: Add FrontdoorSecret

jonasz-lasut opened the pull request:
Fixes #1104

I have:

- [X] Read and followed Crossplane's [contribution process].
- [X] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested
The code had to be tested manually as FrontdoorSecret requires a certificate which is not self-signed and therefore I could not easily generate it with KeyVault Certificate resource + it requires knowledge about Microsoft.AzurefrontDoor-Cdn service principal which may not exist in the Entra Directory (enabled with `New-AzADServicePrincipal -ApplicationId "205478c0-bd83-4e1b-a9d6-db63a3e1e1c8"` and not like terraform example shows `New-AzADServicePrincipal -ApplicationId "00000000-0000-0000-0000-000000000000"`)

Vaults had names changed from _example_ to  _example-1_ and _example-2_ during manual tests to not encounter not purged vaults between tests

Results of cluster-scoped resource manual tests:
<img width="904" height="407" alt="image" src="https://github.com/user-attachments/assets/207d99ee-2448-43ac-b421-a923a0702178" />

Results of namespace-scoped resource manual tests:
<img width="910" height="397" alt="image" src="https://github.com/user-attachments/assets/3f224082-374b-40ea-a36a-41eb5f4785a1" />

[contribution process]: https://git.io/fj2m9

turkenf commented on examples/namespaced/cdn/v1beta1/frontdoorsecret.yaml:
```diff
@@ -0,0 +1,161 @@
+      create a secret which contains certificate which is not self-signed
+  labels:
+    testing.upbound.io/example-name: example
+  name: example-cdn
```
```suggestion
  name: example-cdn
  namespace: upbound-system
```

turkenf commented on examples/namespaced/cdn/v1beta1/frontdoorsecret.yaml:
```diff
@@ -0,0 +1,161 @@
+      create a secret which contains certificate which is not self-signed
+  labels:
+    testing.upbound.io/example-name: example
+  name: example
```
```suggestion
  name: example
  namespace: upbound-system
```

turkenf commented on examples/namespaced/cdn/v1beta1/frontdoorsecret.yaml:
```diff
@@ -0,0 +1,161 @@
+      create a secret which contains certificate which is not self-signed
+  labels:
+    testing.upbound.io/example-name: example
+  name: example
```
```suggestion
  name: example
  namespace: upbound-system
```

turkenf commented on examples/namespaced/cdn/v1beta1/frontdoorsecret.yaml:
```diff
@@ -0,0 +1,161 @@
+      create a secret which contains certificate which is not self-signed
+  labels:
+    testing.upbound.io/example-name: example
+  name: example
```
```suggestion
  name: example
  namespace: upbound-system
```

jonasz-lasut commented:
Added missing namespace, removed from externalnamenottested and run up to date tests:

Cluster scoped:
<img width="972" height="1061" alt="cluster-scoped-frontdoor-up-to-date" src="https://github.com/user-attachments/assets/ebd24f42-578d-4e75-a388-d82572a43e83" />

Namespace scoped:
<img width="1103" height="905" alt="namespaced-scoped-frontdoor-up-to-date" src="https://github.com/user-attachments/assets/def9e019-dcfa-4d61-b2ed-ac7704766991" />

Showing only status.conditions as status.atProvider shows my environment specific domains etc

turkenf reviewed (approved):
Thank you @jonasz-lasut 🙌
